### PR TITLE
IBP-3950: added endpoint and business logic to merge lots

### PR DESCRIPTION
**Pull request by lik-83**

### CODE REVIEW

File: src/main/java/org/ibp/api/java/impl/middleware/inventory/manager/LotServiceImpl.java [hunk 1]
```diff
@@ -198,9 +192,24 @@ public LotSearchMetadata getLotsSearchMetadata(final LotsSearchDto lotsSearchDto
...
+		extendedLotListValidator.validateClosedLots(lotDtos);
 		final WorkbenchUser loggedInUser = this.securityService.getCurrentlyLoggedInUser();
 		lotService.closeLots(loggedInUser.getUserid(), lotDtos.stream().map(ExtendedLotDto::getLotId).collect(Collectors.toList()));
 	}
 
+	@Override
+	public void mergeLots(final String keepLotUUID, final LotsSearchDto lotsSearchDto) {
+		final List<ExtendedLotDto> lotDtos = this.lotService.searchLots(lotsSearchDto, null);
+		this.lotMergeValidator.validate(keepLotUUID, lotDtos);
+		lotsSearchDto.setLotIds(lotDtos.stream().map(LotDto::getLotId).collect(Collectors.toList()));
```
Comment (clarysabel) [inline src/main/java/org/ibp/api/java/impl/middleware/inventory/manager/LotServiceImpl.java:204]: seems like this assignment is not needed @lik-83

Comment (lik-83) [reply src/main/java/org/ibp/api/java/impl/middleware/inventory/manager/LotServiceImpl.java]: Done!

File: src/test/java/org/ibp/api/java/impl/middleware/inventory/LotMergeValidatorTest.java [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+import java.util.UUID;
+
+import static org.hamcrest.CoreMatchers.hasItem;
+import static org.hamcrest.CoreMatchers.instanceOf;
+import static org.hamcrest.MatcherAssert.assertThat;
+
+@RunWith(MockitoJUnitRunner.class)
+public class LotMergeValidatorTest {
+
+	private static final String LOT_KEEP_UUID = "lotUUID";
```
Comment (clarysabel) [inline src/test/java/org/ibp/api/java/impl/middleware/inventory/LotMergeValidatorTest.java:24]: I prefer to use RandomStringUtils, to we ensure the tests pass with any input @lik-83 
In general, again add finals

File: src/test/java/org/ibp/api/java/impl/middleware/inventory/manager/LotServiceImplTest.java [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+import org.mockito.MockitoAnnotations;
+import org.mockito.junit.MockitoJUnitRunner;
+
+import java.util.Arrays;
+import java.util.List;
+
+@RunWith(MockitoJUnitRunner.class)
+public class LotServiceImplTest {
+
+    private static final Integer USER_ID = 1234;
```
Comment (clarysabel) [inline src/test/java/org/ibp/api/java/impl/middleware/inventory/manager/LotServiceImplTest.java:23]: I prefer the random values @lik-83

File: src/test/java/org/ibp/api/java/impl/middleware/inventory/LotMergeValidatorTest.java (hunk 2 above)
Comment (lik-83) [reply src/test/java/org/ibp/api/java/impl/middleware/inventory/LotMergeValidatorTest.java]: Done!

File: src/test/java/org/ibp/api/java/impl/middleware/inventory/manager/LotServiceImplTest.java (hunk 3 above)
Comment (lik-83) [reply src/test/java/org/ibp/api/java/impl/middleware/inventory/manager/LotServiceImplTest.java]: Done!

**Pull request merged**